File: eau10.py




"""
Créez un programme qui affiche toutes les valeurs comprises entre deux nombres dans l’ordre croissant. Min inclus, max exclus.


Exemples d’utilisation :
$> python exo.py 20 25
20 21 22 23 24


$> python exo.py 25 20
20 21 22 23 24

$> python exo.py hello
error

Afficher error et quitter le programme en cas de problèmes d’arguments.

"""
# demander a l'utilisateur de rentré un chiffre 
# l'utilisateur doit voir affichée toutes les valeurs comprise entre deux nombres dans l'ordre croissant min in clus et max inclus 
# faire  une boucle qui aaffiche toute les valeurs entre deux nombre dans l'ordre croissant ou une condition ou les deux 



def demander_nombre (nb_min, nb_max):
    nombre_int=0
    
    for i in range(nb_min, nb_max):
        print(i)

NOMBRE_MAX=input("entrez un nombre min: ")
NOMBRE_MIN=input("entrez un nombre max: ")

print(demander_nombre(int(NOMBRE_MIN), int(NOMBRE_MAX)))

File: eau10.js
/* 
Créez un programme qui affiche le premier index d’un élément recherché dans un tableau. Le tableau est constitué de tous les arguments sauf le dernier. L’élément recherché est le dernier argument. Afficher -1 si l’élément n’est pas trouvé.

Exemples d’utilisation :
$> python exo.py 
6


$> python exo.py test test test
0

$> python exo.py test boom
-1

Afficher error et quitter le programme en cas de problèmes d’arguments.
 */

//parsing
function getArguments(){
    const arguments = process.argv.slice(2)
    return arguments 
    

}

//Gestion d'erreur
function isvalidArguments(validArgumentLength){
    if (!validArgumentLength){
        console.log("ce script nécéssite minimum deux arguments")
        return validArgumentLength
    }

}


function findFirstIndex(){
    const arguments = getArguments();

    const lastIndex = arguments.length - 1
    const lastArg = arguments[lastIndex]
   
  
    for (let i = 0; i < lastIndex; i++){
        if (arguments[i] === lastArg){
            return i; //trouvé 
        }
    }
    return -1; // non trouvé

}



//Resolution 
function getfindFirstIndex(){
    const arguments = getArguments()

    const validArgumentLength = arguments.length === 0
    
    if (!validArgumentLength(arguments) ){
        return
    }
    const findFirstIndex = getfindFirstIndex((arguments))

    return findFirstIndex
}

//Affichage
function displayIndex(){
    console.log(findFirstIndex());

}

displayIndex();































